src/modules/facility-loan-transaction/facility-loan-transaction.service.ts

```diff
@@ -0,0 +1,50 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs-authentication/acbs-authentication.service';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+
+import { AcbsFacilityLoanTransactionService } from '../acbs/acbs-facility-loan-transaction.service';
+import { GetFacilityLoanTransactionResponseItem } from './dto/get-loan-transaction-response.dto';
+
+@Injectable()
+export class FacilityLoanTransactionService {
+  constructor(
+    private readonly acbsAuthenticationService: AcbsAuthenticationService,
+    private readonly acbsFacilityLoanTransactionService: AcbsFacilityLoanTransactionService,
+    private readonly dateStringTransformations: DateStringTransformations,
+  ) {}
+
+  async getLoanTransactionsByBundleIdentifier(bundleIdentifier: string): Promise<GetFacilityLoanTransactionResponseItem> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+    const loanTransaction = await this.acbsFacilityLoanTransactionService.getLoanTransactionByBundleIdentifier(bundleIdentifier, idToken);
+    const loan = loanTransaction.BundleMessageList[0];
+    if (loan.$type !== 'NewLoanRequest') {
+      throw new BadRequestException('Bad request', 'The provided bundleIdentifier does not correspond to a loan transaction.');
+    }
+    const pacAccrual = loan.AccrualScheduleList.find((accrual) => accrual.AccrualCategory.AccrualCategoryCode === 'PAC01');
+    const ctlAccrual = loan.AccrualScheduleList.find((accrual) => accrual.AccrualCategory.AccrualCategoryCode === 'CTL01');
```

Comment:
Do we need CTL01 to be a constant?

Reply:
Changed both PAC01 and CTL01 to use constants.